src/spaces/openstreetmap.jl

```diff
@@ -951,6 +951,26 @@ function Agents.nearby_ids(
     return nearby
 end
 
+function Agents.nearby_ids(
+        pos::Tuple{Float32, Float32},
+        model::ABM{<:OpenStreetMapSpace},
+        distance::Real,
+        args...;
+        kwargs...
+    )
+    nearby = []
+    # need to create an iterator for all agents' lonlat positions as Float32
+    # the Float32 conversion is needed for inspection in visualization extension
+    # TODO maybe track agents' lonlat positions as Float32 inside the space itself?
+    ids_lonlat32 = (Pair(id, Float32.(lonlat(model[id], model))) for id in allids(model))
+    for (id, apos) in ids_lonlat32
+        if sqrt(sum(abs2.(pos .- apos))) <= distance # simple euclidean distance
+            push!(nearby, id)
+        end
+    end
+    return nearby
+end
```

Comment:
```suggestion
    # need to create an iterator for all agents' lonlat positions as Float32
    # the Float32 conversion is needed for inspection in visualization extension
    # TODO maybe track agents' lonlat positions as Float32 inside the space itself?
    ids_lonlat32 = ((id, Float32.(lonlat(model[id], model)) for id in allids(model))
    nearby = (id for (id, apos) in ids_lonlat32 if euclidean_distance(pos, apos, model) <= distance)
    return nearby
end
```

Reply:
I'm also not sure why you need the float32 conversion if you only return nearby ids from this function?

Reply:
Our `lonlat` function returns Float64 values and the positions in Makie plots are `Float32`. We have to convert this here because of the lower precision that comes from the mouse position. Otherwise we would need to run some `isapprox` on the `Float64` `lonlat` positions and to be honest, converting them to `Float32` seemed like the way simpler approach to me.

Reply:
Your suggestion to use the `euclidean_distance` method is totally fine with me but we would need to define that first for `OSMSpace`. Do we need this method somewhere else as well (or maybe in the future)? If so, I would also prefer to have it as a specialised `euclidean_distance` method.